### PR TITLE
Apply style fixes and correct timestamps to dynatrace exporter

### PR DESCRIPTION
**Pull request by dyladan**
**Description:**
- Fix: Timestamp conversion from nano to milli fixed
- Enhancement: Create HTTP request with context
- Fix: Include prefix in metric name length
- Style fixes

**Testing:** updated tests to reflect serialization fixes

Co-Authored by: @PeterFeicht

### CODE REVIEW

**Comment by dyladan** [comment]
@bogdandrutu the failing test is unrelated to the changes I made. can you retrigger the tests?

File: exporter/dynatraceexporter/go.mod [hunk 1]
```diff
@@ -4,6 +4,6 @@ go 1.14
 
 require (
 	github.com/stretchr/testify v1.6.1
-	go.opentelemetry.io/collector v0.15.1-0.20201130171404-7c281055f7e4
+	go.opentelemetry.io/collector v0.15.1-0.20201125171618-60498105d42f
```
Comment (bogdandrutu) [inline exporter/dynatraceexporter/go.mod:7]: Why this change?

Comment (dyladan) [reply exporter/dynatraceexporter/go.mod]: Wasn't intentional. Reverted.

**Review by bogdandrutu — APPROVED** [review]
